### ISSUE TITLE
Adding custom parameters to the custom evaluation and loss functions

### ISSUE THREAD

**Issue by epslong** [dmlc/xgboost#2527]
Hi, 

I'm trying to write custom evaluation and loss functions for my problem context. In this case, I am interested in using quantile loss as the evaluation function. In this case, the evaluation (loss function as well) would need to take external parameters (shown in the example below). I haven't had success implementing below. Appreciate if someone has similar situations and can comment. Thanks!

  qEvalError = function(preds, dfTrain, quant=0.5,...){
    labels = getinfo(dfTrain,"label")
    temp = labels-preds
    qLoss = quant*temp
    qLoss[which(qLoss<0)] = (1-quant)*(-temp[which(qLoss<0)])
    return(list(metric = "QLoss", value = qLoss))
  }

**Comment by IversonR** [comment]
'quant' is defined as global variable.

**Comment by epslong** [comment]
This works. Thanks!
